sdk/ts/src/gateway/index.ts

```diff
@@ -0,0 +1,62 @@
+import { CID } from 'multiformats'
+import { CarReader } from '@ipld/car'
+import fetch from 'cross-fetch'
+import { verify_block } from '../utils'
+import { Block } from '@ipld/car/reader'
+
+export const DEFAULT_GATEWAY = 'https://gateway.ursa.earth'
+
+/// Gateway provides a simple interface to the Ursa Gateway API.
+export class Gateway {
+    constructor(url?: URL | string) {
+        if (url) {
+            const urlStr =
+                typeof url === 'string' ? url : url.toString().slice(0, -1)
+            if (urlStr.endsWith('/')) {
```

Comment:
wondering if the first `slice(0, -1)` necessary?

Reply:
URL always returns the trailing slash so I removed since we know. I guess its not required so removed the extra slice for you